### ISSUE TITLE
support for flow comment syntax

### ISSUE THREAD

**Issue by tnguyen14** [eslint/eslint#7813]
**What version are you using?**
v2.10.2

**What did you do?**
Flow supports the comment syntax, such as
```js
/*: variable type */
```

**What happened?**
However, writing it as such will cause an error with the `spaced-comment` rule 
```
 Expected space or tab after '/*' in comment. (spaced-comment)
```
**What did you expect to happen?**
Should eslint be able to detect the file using flow (with the header `/* @flow */` and relax this error?

**Comment by alberto** [comment]
According to [this post](https://flowtype.org/blog/2015/02/20/Flow-Comments.html) you can use spaces after the comment token, so it makes sense you use that syntax if you use `spaced-comment`.

> whitespace is ignored after the /* but before the :, ::, or flow-include. So you can write things like

**Comment by nzakas** [comment]
> Should eslint be able to detect the file using flow (with the header /* @flow */ and relax this error?

ESLint doesn't know anything about Flow syntax, and our policy is not to make exceptions for anything that isn't standard JavaScript.

**Comment by kaicataldo** [comment]
@tnguyen14 Has your question been answered? Looks like the rule is behaving as intended and that Flow allows for this style of comment.

**Comment by tnguyen14** [comment]
@alberto thanks for the suggestion. I tried `/* : object */` syntax, but then I see this error:
`  Unexpected token /* :`

@nzakas I agree with that statement. However, I feel that expecting space or tab after `/*` is not really a "standard" language feature, more of a stylistic one?

**Comment by nzakas** [comment]
@tnguyen14 you requested that ESLint look for a Flow comment and then adjust the rule output based on that. That would require ESLint to add an exception that is Flow-specific, and as I said, our policy is not to make such exceptions. 

I'd suggest following up with the Flow folks to get a better understanding of what syntax is allowed and potentially creating your own rule regarding comment spacing that fits your needs.

**Comment by platinumazure** [comment]
I'm not sure we even need to do anything here. @tnguyen14 Have you tried using the [`"markers"` option](http://eslint.org/docs/rules/spaced-comment#markers) in the `spaced-comment` rule? If you simply specify `":"` as a marker, then the rule shouldn't trigger for Flow comments.

**Comment by tnguyen14** [comment]
@platinumazure thanks for the suggestion. I'll try that. I'm using `happiness`, which is a variant of `standard`. What would be the cleanest way to specify that rule without adding it to the top of every file?

**Comment by tnguyen14** [comment]
@nzakas Thanks for the explanation.

What I'm wondering is - why would their syntax not be "allowed"? I don't know of a JS rule that says there must be a space after the comment notation, is there?

**Comment by nzakas** [comment]
@tnguyen14 you should use a config file to enable the rules you want. I can't speak for happiness or standard, but if that doesn't work, you probably want to ask those authors.

**Comment by nzakas** [comment]
> What I'm wondering is - why would their syntax not be "allowed"? I don't know of a JS rule that says there must be a space after the comment notation, is there?

Sorry, I have no idea what you're asking here. I'm not familiar with Flow and ESLint doesn't support it natively, so my suggestion was to ask the Flow team if you have questions about syntax allowances.

**Comment by ilyavolodin** [comment]
@tnguyen14 You problem is basically using off-the-shelf eslint wrapper. Those wrappers are specifically made to be easy to use, but provide no ability to customize configuration.
You have two choices, either you can request a change to `spaced-comments` rule configuration from the happiness team, or you can stop using the wrapper and use their [shareable eslint config](https://www.npmjs.com/package/eslint-config-happiness) and eslint itself without the wrapper.

In both cases this is not an issue with ESLint itself, but with the wrapper application. So closing this issue, since there's nothing we can do about this, and @platinumazure already provided a correct workaround.
